src/libcore/option.rs

```diff
@@ -1188,6 +1188,11 @@ impl<A, V: FromIterator<A>> FromIterator<Option<A>> for Option<V> {
                     None => None,
                 }
             }
+
+            #[inline]
+            fn size_hint(&self) -> (usize, Option<usize>) {
+                self.iter.size_hint()
```

Comment:
This looks incorrect since this iterator can yield fewer items than `self.iter`. See `std::iter::Filter::size_hint`.

Reply:
Oops, forgot to commit this file. I'll commit and push it tomorrow.